contracts/access/AccessControlDefaultAdminRules.sol

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts (last updated v4.8.0) (access/AccessControlDefaultAdminRules.sol)
+
+pragma solidity ^0.8.0;
+
+import "./AccessControl.sol";
+import "../utils/math/SafeCast.sol";
+import "../interfaces/IERC5313.sol";
+
+/**
+ * @dev Extension of {AccessControl} that allows specifying special rules to manage
+ * the `DEFAULT_ADMIN_ROLE` holder, which is a sensitive role with special permissions
+ * over other roles that may potentially have privileged rights in the system.
+ *
+ * If a specific role doesn't have an `adminRole` assigned, the holder of the
+ * `DEFAULT_ADMIN_ROLE` will have the ability to manage it, as determined by the
+ * function {getRoleAdmin}'s default value (`address(0)`).
+ *
+ * This contract implements the following risk mitigations on top of the {AccessControl} implementation:
+ *
+ * * Only one account holds the `DEFAULT_ADMIN_ROLE` at every time after construction except when it's renounced.
+ * * Enforce a 2-step process to transfer the `DEFAULT_ADMIN_ROLE` to another account.
+ * * Enforce a configurable delay between the two steps, with the ability to cancel in between.
+ *   - Even after the timer has passed to avoid locking it forever.
+ * * The `DEFAULT_ADMIN_ROLE` 's admin can be only itself.
+ *
+ * Example usage:
+ *
+ * ```solidity
+ * contract MyToken is AccessControlDefaultAdminRules {
+ *   constructor() AccessControlDefaultAdminRules(
+ *     60 * 60 * 24, // 3 days
+ *     _msgSender() // Explicit initial `DEFAULT_ADMIN_ROLE` holder
+ *    ) {}
+ *}
+ * ```
+ *
+ * NOTE: The `delay` is only configurable in the constructor to avoid issues related with handling
+ * delay management during the transfer is pending to be completed.
+ *
+ * _Available since v4.9._
+ */
+abstract contract AccessControlDefaultAdminRules is IERC5313, AccessControl {
+    uint48 private immutable _delay;
+
+    address private _currentDefaultAdmin;
+    address private _pendingDefaultAdmin;
+
+    uint48 private _defaultAdminTransferDelayedUntil;
+
+    /**
+     * @dev Emitted when a `DEFAULT_ADMIN_ROLE` transfer is started, setting `newDefaultAdmin`
+     * as the next default admin, which will have rights to claim the `DEFAULT_ADMIN_ROLE`
+     * after `defaultAdminTransferDelayedUntil` is met.
+     */
+    event DefaultAdminRoleChangeStarted(address indexed newDefaultAdmin, uint48 defaultAdminTransferDelayedUntil);
+
+    /**
+     * @dev Sets the initial values for {delay} in seconds and {defaultAdmin}.
+     *
+     * The `delay` value is immutable. It can only be set at the constructor.
+     */
+    constructor(uint48 delay_, address initialDefaultAdmin) {
+        _delay = delay_;
+        _grantRole(DEFAULT_ADMIN_ROLE, initialDefaultAdmin);
+    }
+
+    /**
+     * @dev Returns the delay between each `DEFAULT_ADMIN_ROLE` transfer.
+     */
+    function delay() public view returns (uint48) {
```

Comment:
any reason why this one is not virtual ?

Reply:
Not in particular, we were using the raw private value (`_delay`) in `beginDefaultAdminTransfer` but I double-checked and it makes sense to have it virtual.

Fixed (and also solving since the recommendation is clear)